src/Client.js

```diff
@@ -275,27 +275,26 @@ export default class Client {
 
     // sendCommand :: Command -> Number -> Promise Command
     sendCommand(command, timeout = this._application.commandTimeout) {
-        var commandPromise = Promise.race([
+        const commandPromise = Promise.race([
             new Promise((resolve, reject) => {
                 this._commandResolves[command.id] = (c) => {
-                    if (!c.status)
-                        return;
-
-                    if (c.status === Lime.CommandStatus.SUCCESS) {
+                    if (!c.status){
+                        reject(new ClientError('Command received without a status'));
+                    } else if (c.status === Lime.CommandStatus.SUCCESS) {
                         resolve(c);
-                    }
-                    else {
+                    } else {
                         const cmd = JSON.stringify(c);
                         reject(new ClientError(cmd));
                     }
 
                     delete this._commandResolves[command.id];
                 };
             }),
-            new Promise((_, reject) => {
+            new Promise((resolve, reject) => {
                 setTimeout(() => {
-                    if (!this._commandResolves[command.id])
-                        return;
+                    if (!this._commandResolves[command.id]) {
```

Comment:
Why do you need to check it?

Reply:
I just added the bracket, but this if is used to guarantee the promise will only be reject for commands with timeout

Reply:
That if statement seems to be impossible to be true.

Reply:
But anyway, I don't think that returning a string can be a good thing, think that consumers are expecting a specific type, in the case a lime envelope, returning a different type can cause problems.